crates/pera-core/src/dwallet_mpc/mpc_session.rs

```diff
@@ -143,12 +132,132 @@ impl DWalletMPCSession {
+                            .encrypted_secret_share_and_proof
+                            .clone(),
+                        encryption_key: event_data.encryption_key.clone(),
+                        encryption_key_id: event_data.encryption_key_id.clone(),
+                        initiator: event_data.initiator.clone(),
+                        signed_public_share: event_data.signed_public_share.clone(),
+                        encryptor_ed25519_pubkey: event_data.encryptor_ed25519_pubkey.clone(),
+
+                        // Fields not relevant for verification; passing empty values.
+                        dwallet_id: ID::new(ObjectID::new([0; 32])),
+                        session_id: ID::new(ObjectID::new([0; 32])),
+                    })?;
+                }
+                Ok(result)
+            }
+            MPCRound::PresignFirst(..) => {
+                let public_input = bcs::from_bytes(&self.public_input)?;
+                crate::dwallet_mpc::advance::<PresignFirstParty>(
+                    session_id,
+                    self.party_id,
+                    &self.weighted_threshold_access_structure,
+                    self.pending_messages.clone(),
+                    public_input,
+                    (),
+                )
+            }
+            MPCRound::PresignSecond(..) => {
+                let public_input = bcs::from_bytes(&self.public_input)?;
+                crate::dwallet_mpc::advance::<PresignSecondParty>(
+                    session_id,
+                    self.party_id,
+                    &self.weighted_threshold_access_structure,
+                    self.pending_messages.clone(),
+                    public_input,
+                    (),
+                )
+            }
+            MPCRound::Sign(..) => {
+                let public_input = bcs::from_bytes(&self.public_input)?;
+                crate::dwallet_mpc::advance::<SignFirstParty>(
+                    session_id,
+                    self.party_id,
+                    &self.weighted_threshold_access_structure,
+                    self.pending_messages.clone(),
+                    public_input,
+                    self.decryption_share.clone(),
+                )
+            }
+            MPCRound::NetworkDkg(key_scheme, _) => advance_network_dkg(
+                session_id,
+                &self.weighted_threshold_access_structure,
+                self.party_id,
+                &self.public_input,
+                key_scheme,
+                self.pending_messages.clone(),
+                bcs::from_bytes(
+                    &self
+                        .private_input
+                        .clone()
+                        .ok_or(DwalletMPCError::MissingMPCPrivateInput)?,
+                )?,
+            ),
+            MPCRound::EncryptedShareVerification(verification_data) => {
+                match verify_encrypted_share(verification_data) {
```

Comment:
I wanted to ask: why is this here twice, once after DKG and once by its own. I know the design so I understood that this is for dWallet transfer where the other is for dWallet creation: but other won't! This must be documented